google-ads-examples/src/main/java/com/google/ads/googleads/examples/misc/UploadImage.java

```diff
@@ -0,0 +1,125 @@
+import com.google.protobuf.BytesValue;
+import com.google.protobuf.ByteString;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+/** This example uploads an image. */
+public class UploadImage {
+
+  private static class AddAdGroupParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    private Long customerId;
+  }
+
+  public static void main(String[] args) throws IOException {
+    AddAdGroupParams params = new AddAdGroupParams();
+    if (!params.parseArguments(args)) {
+
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("INSERT_CUSTOMER_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+        "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new UploadImage().runExample(googleAdsClient, params.customerId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+        "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+        gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @throws GoogleAdsException if an API request failed with one or more service errors.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId)
+      throws IOException {
+
+    byte[] imageData = ByteStreams.toByteArray(new URL("https://goo.gl/3b9Wfh").openStream());
+
+    MediaImage image =
+      MediaImage.newBuilder()
+        .setData(BytesValue.of(ByteString.copyFrom(imageData)))
+        .build();
+
+    MediaFile file =
+      MediaFile.newBuilder()
+        .setName(StringValue.of("Ad Image"))
+        .setType(MediaType.IMAGE)
+        .setSourceUrl(StringValue.of("https://goo.gl/3b9Wfh"))
+        .setImage(image)
+        .build();
+
+    List<MediaFileOperation> operations = new ArrayList<>();
```

Comment:
Can you replace this with:

MediaFileOperation op = ...

mediaFileServiceClient.mutateMediaFiles(Long.toString(customerId), Arrays.asList(op))

please. A couple of reasons for this, primarily that it's consistent with our existing examples, and also that new ArrayList() will allocate an array of size 32 to hold on operation.